src/target/adiv5.c

```diff
@@ -581,19 +581,22 @@ static void adiv5_component_probe(
 		designer_code = (pidr & PIDR_JEP106_CONT_MASK) >> (PIDR_JEP106_CONT_OFFSET - 8U) |
 			(pidr & PIDR_JEP106_CODE_MASK) >> PIDR_JEP106_CODE_OFFSET;
 
-		if (designer_code == JEP106_MANUFACTURER_ERRATA_STM32WX || designer_code == JEP106_MANUFACTURER_ERRATA_CS) {
-			/**
-			 * see 'JEP-106 code list' for context, here we are aliasing codes that are non compliant with the
-			 * JEP-106 standard to their expected codes, this is later used to determine the correct probe function.
-			 */
-			DEBUG_WARN("Patching Designer code 0x%03" PRIx16 " -> 0x%03u\n", designer_code, JEP106_MANUFACTURER_STM);
-			designer_code = JEP106_MANUFACTURER_STM;
-		}
 	} else {
 		/* legacy ascii code */
 		designer_code = (pidr & PIDR_JEP106_CODE_MASK) >> PIDR_JEP106_CODE_OFFSET | ASCII_CODE_FLAG;
 	}
 
+	if (designer_code == JEP106_MANUFACTURER_ERRATA_STM32WX || designer_code == JEP106_MANUFACTURER_ERRATA_CS ||
+		designer_code == JEP106_MANUFACTURER_ERRATA_CS_ASCII) {
+		/**
+         * see 'JEP-106 code list' for context, here we are aliasing codes that are non compliant with the
+         * JEP-106 standard to their expected codes, this is later used to determine the correct probe function.
+         */
+		DEBUG_WARN(
+			"Patching Designer code 0x%03" PRIx16 " -> 0x%03" PRIx16 "\n", designer_code, JEP106_MANUFACTURER_STM);
```

Comment:
Keeping in mind that newlib has some fairly restricted handling of formatting specifiers, it might be a good idea here to just use "%03x" for the format strings as that's perfectly fine with uint16_t's going in..

It does feel weird to us including the `0x` prefix here and below, but we do recognise it creates a certain consistency with the "Please report unknown device" strings, even if we feel as these are always in hexadecimal that saving 4 bytes per string by dropping the prefix is a good space saving which doesn't hamper anything.

Reply:
Removed the "0x" prefix for designer and part id and used "%03x" as format string (again) as suggested.